drivers/hmis/lib/hmis_data_cleanup/util.rb

```diff
@@ -354,5 +354,72 @@ def self.cleanup_dangling_records_from_merge!(klass: Hmis::Hud::CustomCaseNote)
         Rails.logger.info "WARNING: #{remaining_dangling} dangling records remain"
       end
     end
+
+    # Method for restoring enrollments that were deleted in an import.
+    # Probably needs to be called in batches, depending on the size of the enrollment scope.
+    def restore_deleted_enrollments!(enrollments_to_restore, conflict_set: nil, dry_run: false)
+      # Commented-out examples of parameters:
+      # enrollments_to_restore = Hmis::Hud::Enrollment.hmis.only_deleted.where(project_id: 650).where(date_deleted: date)
+      # conflict_set = Hmis::Hud::Enrollment.hmis.where(project_id: 650).pluck(:ProjectID, :PersonalID, :EntryDate).to_set
+
+      valid_personal_ids = Hmis::Hud::Client.hmis.pluck(:personal_id).uniq.to_set
+      personal_ids_from_deleted_records = enrollments_to_restore.pluck(:personal_id).uniq
+
+      # Map { old personal id => new personal id } for clients that have been merged
+      old_to_new_personal_id = Hmis::Hud::Client.hmis.with_deleted.where(personal_id: personal_ids_from_deleted_records).map do |client|
+        [client.personal_id, client.reverse_merge_histories.first&.retained_client&.personal_id]
+      end.to_h
+
+      conflicts = 0 # num skipped due to conflict (PersonalID + Entry Date, meaning the enrollment has been re-created since it was deleted)
+      enrollments_where_personal_id_updated = 0 # num enrollments where PersonalID was updated due to a merge
+      enrollments_skipped_due_to_unrecognized_personal_id = 0 # num enrollments skipped because personal ID not found (dnd not merged)
```

Comment:
```suggestion
      enrollments_skipped_due_to_unrecognized_personal_id = 0 # num enrollments skipped because personal ID not found (and not merged)
```